File: MyGame_OC/gameTest/gameTest/GameViewController.m
//
//  GameViewController.m
//  gameTest
//
//  Created by 林 on 2019/10/23.
//  Copyright © 2019 林. All rights reserved.
//

#import "GameViewController.h"
#import "GameScene.h"

@implementation GameViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    // Load the SKScene from 'GameScene.sks'
//    GameScene *scene = (GameScene *)[SKScene nodeWithFileNamed:@"GameScene"];
    
//    SKScene *scene = [[SKScene alloc] initWithSize:CGSizeMake([UIScreen mainScreen].bounds.size.width, [UIScreen mainScreen].bounds.size.height)];
    GameScene *scene = [GameScene sceneWithSize:CGSizeMake(284, 512)];
    // Set the scale mode to scale to fit the window
    scene.scaleMode = SKSceneScaleModeFill;
    
    SKView *skView = (SKView *)self.view;
    
    // Present the scene
    [skView presentScene:scene];
    
    skView.showsFPS = YES;
    skView.showsNodeCount = YES;
}

- (BOOL)shouldAutorotate {
    return YES;
}

- (UIInterfaceOrientationMask)supportedInterfaceOrientations {
    if ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPhone) {
        return UIInterfaceOrientationMaskAllButUpsideDown;
    } else {
        return UIInterfaceOrientationMaskAll;
    }
}

- (BOOL)prefersStatusBarHidden {
    return YES;
}

@end

File: gameTestSwift/gameTestSwift/GameViewController.swift
//
//  GameViewController.swift
//  gameTestSwift
//
//  Created by 林 on 2019/10/29.
//  Copyright © 2019 林. All rights reserved.
//

import UIKit
import SpriteKit
import GameplayKit

class GameViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        if let view = self.view as! SKView? {
            // Load the SKScene from 'GameScene.sks'
//            if let scene = SKScene(fileNamed: "GameScene") {
            let scene = GameScene.init(size: CGSize.init(width: 284, height: 512))
            scene.scaleMode = .aspectFill
            
            // Present the scene
            view.presentScene(scene)
//            if  {
//                // Set the scale mode to scale to fit the window
//              
//            }
            
            view.ignoresSiblingOrder = true
            
            view.showsFPS = true
            view.showsNodeCount = true
        }
    }

    override var shouldAutorotate: Bool {
        return true
    }

    override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
        if UIDevice.current.userInterfaceIdiom == .phone {
            return .allButUpsideDown
        } else {
            return .all
        }
    }

    override var prefersStatusBarHidden: Bool {
        return true
    }
}
